File: TriathlonServerJava/Common/src/main/java/ro/mpp/triathlon/services/ITriathlonServices.java
package ro.mpp.triathlon.services;

import ro.mpp.triathlon.model.Event;
import ro.mpp.triathlon.model.Participant;
import ro.mpp.triathlon.model.Referee;
import java.util.List;

public interface ITriathlonServices {
    Referee login(String username, String password, ITriathlonObserver client) throws Exception;
    void logout(Referee referee, ITriathlonObserver client) throws Exception;
    List<Participant> getParticipantsByEvent(int idEvent) throws Exception;
    List<Event> getAllEvents() throws Exception;
    void addResult(int idReferee, int idParticipant, int points) throws Exception;
}

File: TriathlonCrossPlatform/Triathlon.Services/ITriathlonServices.cs
using System.Collections.Generic;
using Triathlon.Model;

namespace Triathlon.Services
{
    public interface ITriathlonServices
    {
        Referee Login(string username, string password, ITriathlonObserver client);
        void Logout(Referee referee, ITriathlonObserver client);
        List<Participant> GetParticipantsByEvent(int idEvent);
        void AddResult(int idReferee, int idParticipant, int points);
        List<Event> GetAllEvents();
    }
}